### PR TITLE
tests: cmsis_rtos_v1: add negative tests for timer api

### PR DESCRIPTION
**Pull request by ajaykish**
Added negative tests for timer api. It will help to increase
code coverage.

Signed-off-by: Ajay Kishore <ajay.kishore@intel.com>

### CODE REVIEW

File: tests/cmsis_rtos_v1/src/timer.c [hunk 1]
```diff
@@ -97,3 +103,28 @@ void test_timer(void)
...
+	/*Stop the timer before start */
+	status = osTimerStop(id);
+	zassert_true(status == osErrorResource, "error while stoping non active timer");
+
+	timerDelay = ONESHOT_TIME;
+	status = osTimerStart(id, timerDelay);
+	zassert_true(status == osOK, "error starting one-shot timer");
+
+	osDelay(timerDelay + 100);
+
```
Comment (nniranjhana) [inline tests/cmsis_rtos_v1/src/timer.c]: One more negative test: Stop an inactive timer and assert for `osErrorResource`

Comment (ajaykish) [reply tests/cmsis_rtos_v1/src/timer.c]: @nniranjhana In line 117-118, test is stopping an inactive timer, and checking the status. When we create a timer using osTimerCreate(), it sets the timer status to NOT ACTIVE state. Timer status will move to ACTIVE state when we start the timer using  osTimerStart().

Comment (nniranjhana) [reply tests/cmsis_rtos_v1/src/timer.c]: @ajaykish Yes, you are right, sorry I missed that!

**Review by rgundi — CHANGES_REQUESTED** [review]
I don't see a need for separate functions to execute these test cases. You can do these in the existing functions itself.

**Comment by ajaykish** [comment]
@rgundi Moved all changes in the existing function.

**Review by rgundi — APPROVED** [review]
LGTM

**Review by nashif — APPROVED** [review]

**Pull request merged**